CONTRIBUTING.md

```diff
@@ -1,115 +1,136 @@
+- Perform a [search](https://github.com/Ekumen-OS/beluga/issues) to see if the feature has already been requested. If it has, add a comment to the existing issue instead of opening a new one.
+- Find out whether your idea fits with the scope and aims of the project. It's up to you to make a strong case to convince the project's developers of the merits of this feature.
+
+<!-- omit in toc -->
+#### How Do I Submit a Good Feature Request?
+
+Feature requests are tracked as [GitHub issues](https://github.com/Ekumen-OS/beluga/issues).
+
+- Use a **clear and descriptive title** for the issue to identify the suggestion.
+- Provide a **step-by-step description of the suggested feature** in as many details as possible.
+- **Describe the current behavior** and **explain which behavior you expected to see instead** and why. At this point you can also tell which alternatives do not work for you.
+- **Explain why this feature would be useful** to most Beluga users. You may also want to point out the other projects that solved it better and which could serve as inspiration.
+
+### Your First Code Contribution
+
+This section guides you through contributing code to Beluga. Following these guidelines will help maintainers and ensure your contributions are reviewed and eventually accepted.
+
+#### A Primer On Project Workflow
+
+This projects adopts a [feature branch workflow](https://about.gitlab.com/topics/version-control/what-is-git-workflow/#feature-branching-git-workflow) ([forking workflow](https://about.gitlab.com/topics/version-control/what-is-git-workflow/#forking-git-workflow) for contributors that are not maintainers), with pull requests for code integration. Every code contribution must be associated to a [feature request](#requesting-features) or [bug report](#reporting-bugs) with enough consensus and evidence to move forward, signaled with `needs-work` and `needs-fix` labels respectively. Contributors must sign-off each commit by adding a `Signed-off-by: ...` line to every commit message to certify that they have the right to submit the code they are contributing to the project according to the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). Pull requests must have been reviewed and approved at least once to be merged.
+
+#### How Do I Submit a Good Code Contribution?
+
+- **Pull latest `main` branch**. Clone repository (or repository fork) locally if you haven't yet.
+- **Branch from `main` branch**. Use a descriptive name for your feature branch. Make sure to prefix it with the associated issue number.
+- **Work on your contribution**. See [instructions](GETTING_STARTED.md) on how to get started with Beluga development.
+- **Lint and test your changes**. For bug fixes, make sure regression tests are included.
+- **Document your changes as needed**. For new features, make sure added functionality is clearly documented.
+- **Push your feature branch**. To upstream repository (or repository fork).
```

Comment:
Current settings shouldn't allow someone who is not a contributor to push to this repository.
Are you saying we should make direct contributors create forks too? I don't think that's necessary.

Reply:
> Current settings shouldn't allow someone who is not a contributor to push to this repository.

Ok, that's what I was proposing. In that case what repository does "Push your feature branch. To upstream repository" refer to? As a user forking this repo, upstream to me means this repo, not my fork (which would by default be origin). If we don't allow non-contributors to push here, then most people who read this will not be able to push to upstream.

Reply:
Ok, I changed the instructions in https://github.com/Ekumen-OS/beluga/pull/186/commits/5d1f4de22db1ad4e58d693a2758bc6d9d3ce2bd9.

Internal contributors can create forks if they want, but I think there is no need to enforce it internally just yet (pushing a branch to the repo if you have access is allowed for now).